modules/system/assets/js/snowboard/ajax/handlers/AttributeRequest.js

```diff
@@ -0,0 +1,322 @@
+/**
+ * Enable Data Attributes API for AJAX requests.
+ *
+ * This is an extension of the base AJAX functionality that includes handling of HTML data attributes for processing
+ * AJAX requests. It is separated from the base AJAX functionality to allow developers to opt-out of data attribute
+ * requests if they do not intend to use them.
+ *
+ * @copyright 2021 Winter.
+ * @author Ben Thomson <git@alfreido.com>
+ */
+class AttributeRequest extends Snowboard.Singleton {
+    /**
+     * Listeners.
+     *
+     * @returns {Object}
+     */
+    listens() {
+        return {
+            ready: 'ready',
+            ajaxSetup: 'onAjaxSetup'
+        };
+    }
+
+    /**
+     * Ready event callback.
+     *
+     * Attaches handlers to the window to listen for all request interactions.
+     */
+    ready() {
+        this.attachHandlers();
+        this.disableDefaultFormValidation();
+    }
+
+    /**
+     * Dependencies.
+     *
+     * @returns {string[]}
+     */
+    dependencies() {
+        return ['request', 'jsonParser'];
```

Comment:
How are dependencies handled if someone replaces the request class with their own implementation but still wants to be able to use the AttributeRequest class?

Reply:
Dependencies work simply by checking if a plugin with the given name(s) are added. In the case above, `AttributeRequest` is expected a plugin named `request` and a plugin named `jsonParser` to exist, because there's no real way to make a "soft" reference to a class that may or may not exist in JS.

For your particular case, the developer would have two options. They could name their custom Request implementation as "request" which would satisfy the dependency above, or they could extend the `AttributeRequest` class in their own implementation and override the dependencies.